lib/pos_blockchain.go

```diff
@@ -534,25 +659,38 @@ func (bc *Blockchain) canCommitGrandparent(currentBlock *BlockNode) (_grandparen
 // CommittedBlockStatus of the block and flushes the view after connecting the block
 // to the DB and updates relevant badger indexes with info about the block.
 func (bc *Blockchain) commitBlock(blockHash *BlockHash) error {
-	block, exists := bc.uncommittedBlocksMap[*blockHash]
-	if !exists {
-		return errors.Errorf("commitBlock: Block %v not found in uncommitted blocks map", blockHash.String())
-	}
-	// block must be in the best chain. we grab the block node from there.
+	// FIXME(diamondhands): I didn't review this too closely yet. Let's make sure
+	// we're on the same page on ProcessBlock first. This part shouldn't be hard
+	// once the meat of ProcessBlockPoS is done.
+	//
+	// Notice the blockIndex should always have the block we're looking for, and the
+	// block should always be persisted to Badger by the time we hit commitBlock (yay).
+
+	// Block must be in the best chain. we grab the block node from there.
 	blockNode, exists := bc.bestChainMap[*blockHash]
 	if !exists {
 		return errors.Errorf("commitBlock: Block %v not found in best chain map", blockHash.String())
 	}
+	block, err := GetBlock(blockHash, bc.db, bc.snapshot)
+	if err != nil {
+		return errors.Wrapf(err, "commitBlock: Problem getting block from db %v", blockHash.String())
+	}
 	// Connect a view up to the parent of the block we are committing.
 	utxoView, err := bc.getUtxoViewAtBlockHash(*block.Header.PrevBlockHash)
 	if err != nil {
 		return errors.Wrapf(err, "runCommitRuleOnBestChain: Problem initializing UtxoView: ")
 	}
 	// Get the full uncommitted block from the uncommitted blocks map
-	grandParentBlock, exists := bc.uncommittedBlocksMap[*blockHash]
+	// FIXME: This is using blockHash when it should use a parent hash or something.
+	// It was like this when I got here.
+	grandParentBlockNode, exists := bc.blockIndex[*blockHash]
```

Comment:
This is just a typo. We call "commitBlock" on the grandparent block node if the commit rule tells us to. I just forgot to update the name of the variable.